src/Components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { useTodos, useTodosFilter } from '../TodoContext/TodoContext';
+import { Status } from '../../types/Status';
+
+export const TodoList: React.FC = () => {
+  const todos: Todo[] = useTodos();
+  const { filter } = useTodosFilter();
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    switch (filter) {
+      case Status.Active:
+        setVisibleTodos(todos.filter((todo) => !todo.completed));
+        break;
+      case Status.Completed:
+        setVisibleTodos(todos.filter((todo) => todo.completed));
+        break;
+      default:
+        setVisibleTodos(todos);
+    }
+  }, [filter, todos]);
```

Comment:
Refactor - instead of using useEffect + useState - create function that will return visible todos
```suggestion
  const visibleTodos = getVisibleTodos(todos, filter);
```